processor/batch-processor/src/run.ts

```diff
@@ -74,18 +89,32 @@ class Processor<B extends BlockBase, S> {
     }
 
     async run(): Promise<void> {
-        let state = await this.db.connect()
+        let state = await this.getDatabaseState()
+
+        // remove all hot block to start from the finalized head
+        state = {...state, top: []}
```

Comment:
It is not polite to loose progress.

Reply:
> It is not polite to loose progress.

Maybe, but it's just easier to do it in this way, rather than check all "top" blocks